### PR TITLE
Add UI to show duplicate artifacts

### LINKED ISSUE #1110: Duplicate detection UI
I'm not sure how to ask questions since the wiki page doesn't allow me to ask how to do things.

Is there a button or way to detect and delete duplicate artifacts? I used the ArtiScan thing but it made duplicates and almost all my artifacts were duplicated (ㄒoㄒ)

Thank you! And have a great day :)

### PR DESCRIPTION
**Pull request by frzyc**
Resolves #1110
image

### CODE REVIEW

File: apps/frontend/src/app/PageArtifact/DupModal.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  const { t } = useTranslation('artifact')
+  const { database } = useContext(DatabaseContext)
+  const [dbDirty, setDBDirty] = useForceUpdate()
+  useEffect(() => database.arts.followAny(setDBDirty), [setDBDirty, database])
+
+  const dupList = useMemo(() => {
+    const dups = dbDirty && ([] as Array<Array<string>>)
+    let artKeys = database.arts.keys
+
+    while (artKeys.length !== 0) {
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: Why this over `for (const artKey of artKeys)` or `artKeys.forEach()`?

File: apps/frontend/src/app/PageArtifact/DupModal.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    return dups
+  }, [database, dbDirty])
+  return (
+    <Stack spacing={2}>
+      {dupList.map((dups) => (
+        <CardDark key={dups.join()} sx={{ overflowX: 'scroll' }}>
+          <CardContent sx={{ display: 'flex', gap: 1 }}>
+            {dups.map((dup) => (
+              <Box key={dup} sx={{ minWidth: 300 }}>
+                <ArtifactCard
```
Comment (nguyentvan7) [inline apps/frontend/src/app/PageArtifact/DupModal.tsx:74]: Clicking the X in the autocomplete of a duplicated artifact makes the editor UI show up for some reason
image

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/PageArtifact/DupModal.tsx (hunk 1 above)
Comment (nguyentvan7) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: How? Also can you add a comment then

Comment (frzyc) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: `artKeys` is a array of ids that hasnt gotten any dups. when dups are found, all occurence gets removed from the list.

Comment (nguyentvan7) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: Maybe I'm missing something, I don't see how that is happening with [Array.shift()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift)

Comment (frzyc) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: `Array.shift` removes the 1st element of the array and returns it. 
If dups are detected from that element, those are removed as well.

Comment (nguyentvan7) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:49]: oh i totally missed line 63 :)

File: apps/frontend/src/app/PageArtifact/DupModal.tsx (hunk 2 above)
Comment (nguyentvan7) [reply apps/frontend/src/app/PageArtifact/DupModal.tsx:74]: * Open duplicate artifacts
* click edit button for an artifact
* close editor
* click X on autocomplete
* editor will reopen

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**